packages/expo/src/launch/withRestoration.tsx

```diff
@@ -0,0 +1,76 @@
+import Constants, { ExecutionEnvironment } from 'expo-constants';
+import * as React from 'react';
+import { Platform } from 'react-native';
+
+import DevLoadingView from '../environment/DevLoadingView';
+
+export type InitialProps = {
+  exp: {
+    notification?: any;
+    errorRecovery?: any;
+    manifestString?: string;
+    [key: string]: any;
+  };
+  shell?: boolean;
+  shellManifestUrl?: string;
+  [key: string]: any;
+};
+
+const isDevLoadingDisabled =
+  !__DEV__ ||
```

Comment:
I comment more on this later, but IIRC Terser used to not be smart enough to propagate the value of `__DEV__` through multiple expressions and remove dead code. It required `if (__DEV__)` checks. Would be good to double-check that the dev code is all removed in production mode (and perhaps go even further to remove the DevLoadingView import when it's not needed).